duckduckgo/macos-browser
# Adds XPC and NetworkProtectionMac

Task/Issue URL: https://app.asana.com/0/0/1205672018528722/f

## Description

This PR does two important things:
- It fixes a bug with the onboarding flow.
- It adds 3 packages:
    - Intercom: this package adds support for XPC communication
    - NetworkProtectionIPC: this packages adds an IPC implementation for Network Protection based on Intercom
    - NetworkProtectionMac: this is actually not really a new package.  It's just a wrapper package for all of the NetworkProtection packages that are specific for macOS.  Currently these packages are:
        - NetworkProtectionIPC
        - NetworkProtectionUI

## Scope & recommended way to review this PR

This is just an initial break-up of a much larger work.  The idea behind this PR is to merge these modules quickly.

In order to achieve this goal, this PR is mostly about:
- Making sure the App is behaving as usual.
- Making sure the code looks correct.  No testing of the code is really needed right now since it's unwired.  We can do a UX review of the full feature as we move forward.

## Testing:

1. Ensure the onboarding flow is right.
2. Ensure the VPN works normally.
3. Ensure the tests succeed.

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
